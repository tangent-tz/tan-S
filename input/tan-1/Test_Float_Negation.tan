main {
    const floatone := 10.5;
    const floattwo := -5.25;

    # Negation of a positive float
    const negfloatone := -floatone;
    print negfloatone;  # Output: -10.5
    print \n;

    # Negation of a negative float
    const negfloattwo := -floattwo;
    print negfloattwo;  # Output: 5.25
    print \n;

    # Negation of zero
    const zero := 0.0;
    const negzero := -zero;
    print negzero;  # Output: -0.0
    print \n;

    # Negation of a large positive float
    const largepos := 9999.99;
    const neglargepos := -largepos;
    print neglargepos;  # Output: -9999.99
    print \n;

    # Negation of a large negative float
    const largeneg := -8888.88;
    const neglargeneg := -largeneg;
    print neglargeneg;  # Output: 8888.88
    print \n;
}
