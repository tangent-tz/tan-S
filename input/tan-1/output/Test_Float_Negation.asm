        Jump         $$main                    
        DLabel       $eat-location-zero        
        DataZ        8                         
        DLabel       $print-format-integer     
        DataC        37                        %% "%d"
        DataC        100                       
        DataC        0                         
        DLabel       $print-format-boolean     
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-float       
        DataC        37                        %% "%f"
        DataC        102                       
        DataC        0                         
        DLabel       $print-format-character   
        DataC        37                        %% "%c"
        DataC        99                        
        DataC        0                         
        DLabel       $print-format-string      
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-newline     
        DataC        10                        %% "\n"
        DataC        0                         
        DLabel       $print-format-space       
        DataC        32                        %% " "
        DataC        0                         
        DLabel       $print-format-tab         
        DataC        9                         %% "\t"
        DataC        0                         
        DLabel       $boolean-true-string      
        DataC        116                       %% "true"
        DataC        114                       
        DataC        117                       
        DataC        101                       
        DataC        0                         
        DLabel       $boolean-false-string     
        DataC        102                       %% "false"
        DataC        97                        
        DataC        108                       
        DataC        115                       
        DataC        101                       
        DataC        0                         
        DLabel       $errors-general-message   
        DataC        82                        %% "Runtime error: %s\n"
        DataC        117                       
        DataC        110                       
        DataC        116                       
        DataC        105                       
        DataC        109                       
        DataC        101                       
        DataC        32                        
        DataC        101                       
        DataC        114                       
        DataC        114                       
        DataC        111                       
        DataC        114                       
        DataC        58                        
        DataC        32                        
        DataC        37                        
        DataC        115                       
        DataC        10                        
        DataC        0                         
        Label        $$general-runtime-error   
        PushD        $errors-general-message   
        Printf                                 
        Halt                                   
        DLabel       $errors-int-divide-by-zero 
        DataC        105                       %% "integer divide by zero"
        DataC        110                       
        DataC        116                       
        DataC        101                       
        DataC        103                       
        DataC        101                       
        DataC        114                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$i-divide-by-zero        
        PushD        $errors-int-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $errors-float-divide-by-zero 
        DataC        102                       %% "float divide by zero"
        DataC        108                       
        DataC        111                       
        DataC        97                        
        DataC        116                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$f-divide-by-zero        
        PushD        $errors-float-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $usable-memory-start      
        DLabel       $global-memory-block      
        DataZ        88                        
        Label        $$main                    
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        10.500000                 
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        5.250000                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        16                        
        Add                                    %% negfloatone
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        16                        
        Add                                    %% negfloatone
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        24                        
        Add                                    %% negfloattwo
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        24                        
        Add                                    %% negfloattwo
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        32                        
        Add                                    %% zero
        PushF        0.000000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        40                        
        Add                                    %% negzero
        PushD        $global-memory-block      
        PushI        32                        
        Add                                    %% zero
        LoadF                                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        40                        
        Add                                    %% negzero
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        48                        
        Add                                    %% largepos
        PushF        9999.990000               
        StoreF                                 
        PushD        $global-memory-block      
        PushI        56                        
        Add                                    %% neglargepos
        PushD        $global-memory-block      
        PushI        48                        
        Add                                    %% largepos
        LoadF                                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        56                        
        Add                                    %% neglargepos
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        64                        
        Add                                    %% largeneg
        PushF        8888.880000               
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        72                        
        Add                                    %% neglargeneg
        PushD        $global-memory-block      
        PushI        64                        
        Add                                    %% largeneg
        LoadF                                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        72                        
        Add                                    %% neglargeneg
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        80                        
        Add                                    %% f1
        PushF        4.000000                  
        FNegate      null                      
        PushF        3.000000                  
        Nop                                    
        FMultiply    null                      
        PushF        30.000000                 
        FNegate      null                      
        PushF        15.000000                 
        Nop                                    
        Duplicate                              
        JumpFZero    $$f-divide-by-zero        
        Label        -divide-1-notZero         
        FDivide      null                      
        FAdd         null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        80                        
        Add                                    %% f1
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        Halt                                   
