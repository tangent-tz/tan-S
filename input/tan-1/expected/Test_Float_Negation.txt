Negation of a positive float        : -10.5
Negation of a negative float        : 5.25
Negation of zero                    : -0.0
Negation of a large positive float  : -9999.99
Negation of a large negative float  : 8888.88
